ttnn/cpp/ttnn/operations/others.hpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_eager/tt_dnn/op_library/upsample/upsample_op.hpp"
+
+namespace ttnn {
+namespace operations {
+namespace others {
+
+
+inline ttnn::Tensor upsample(
+    const ttnn::Tensor& input_tensor,
+    //std::variant<float, std::tuple<float, float>, std::tuple<float, float, float>, std::tuple<float, float, float, float>> scale_factor,
+    std::variant<int, std::vector<int>> scale_factor,
```

Comment:
why not change this to use std::variant<std::array<float, 2>, std::array<float, 3>, std::array<float, 4>> and std::visit the helper implementation

Reply:
then you avoid the runtime overhead of the vector size checks

Reply:
Missed this comment earlier. Makes sense.. I'm gonna update.